ros/ros1/Utils.hpp

```diff
@@ -57,12 +57,15 @@ inline auto GetTimestampField(const PointCloud2::ConstPtr msg) {
 
 // Normalize timestamps from 0.0 to 1.0
 inline auto NormalizeTimestamps(const std::vector<double> &timestamps) {
-    const double max_timestamp = *std::max_element(timestamps.cbegin(), timestamps.cend());
-    // check if already normalized
-    if (max_timestamp < 1.0) return timestamps;
+    const auto min_max_it = std::minmax_element(timestamps.cbegin(), timestamps.cend());
+    const double min_timestamp = *min_max_it.first;
+    const double max_timestamp = *min_max_it.second;
+
     std::vector<double> timestamps_normalized(timestamps.size());
     std::transform(timestamps.cbegin(), timestamps.cend(), timestamps_normalized.begin(),
-                   [&](const auto &timestamp) { return timestamp / max_timestamp; });
+                   [&](const auto &timestamp) {
+                       return (timestamp - min_timestamp) / (max_timestamp - min_timestamp);
```

Comment:
Timestamps need to be normalized between 0 and 1. Because we select the middle pose in the scan (t=0.5), if min_timestamp is not zero we are taking a random pose in the middle of the the sweep. With this offset we guaranty that the timestamps are always in a valid range. This as not an impact on our pipeline because everything got shifted by the same amount.